### PR TITLE
[Nav] Flyout Responsive Menu

### LINKED ISSUE #918: [Navbar] Mobile Menu Update
Update mobile menu in DSR and Frontend to match screenshots, included horizontal spacing for SBL Beta banner and content on Filing/Profile pages:
mobile-menu

Example of missing horizontal spacing for SBL Beta banner and content:
Screenshot 2024-10-03 at 6 23 20 AM

Actual screenshot with horizontal spacing:
Screenshot 2024-09-27 at 4 08 48 PM

Frontend task:
- https://github.com/cfpb/sbl-frontend/issues/765

### PR DESCRIPTION
**Pull request by ojbravo**
Mobile menu is working. Functionality matches cfpb.gov.

Closes #918.

Prerequisites:
- [x] Approval/merge of https://github.com/cfpb/design-system-react/pull/380
- [x] Update this branch's deps

### CODE REVIEW

**Comment by billhimmelsbach** [comment]
Heyo @ojbravo! This one is waiting on Natalia's seal of approval right?

**Comment by billhimmelsbach** [comment]
Getting this prepped for deployment too.

**Review by billhimmelsbach — COMMENTED** [review]
A build error on this one, maybe due to the pinned design system react dependency? I'll take a look at it after the next meeting.

**Comment by meissadia** [comment]
> A build error on this one, maybe due to the pinned design system react dependency? I'll take a look at it after the next meeting.

I too am getting a crash when previewing the build locally.  Trying to determine why.

File: package.json
Comment (billhimmelsbach) [inline package.json:42]: Heyo! Going to put this up on Dev again for Natalia, but going to put a hold on it just so we don't forget to remove this/merge it in on the DSR before we merge this PR.

**Review by billhimmelsbach — CHANGES_REQUESTED** [review]

**Review by natalia-fitzgerald — DISMISSED** [review]
@meissadia 
The responsive menu looks great and aligns closely with the cf.gov responsive menu. I will add the items I noticed (which aren't directly captured by this PR) to a separate ticket.

File: package.json
Comment (meissadia) [reply package.json]: Mobile nav PR merged in the [DSR](https://github.com/cfpb/design-system-react/pull/380).  
 
This PR updated to point this dependency back at the `main` branch.

**Comment by meissadia** [comment]
@billhimmelsbach `Build + preview` issues seem to have resolved themselved 🤷🏾 

PR looks good to go.

**Review by billhimmelsbach — APPROVED** [review]
Looks good to go!

**Pull request merged**
